packages/peregrine/lib/talons/Image/useImage.js

```diff
@@ -34,18 +32,14 @@ export const useImage = props => {
         }
     }, [onError]);
 
-    // If we don't have a resourceWidth, use the smallest resource size.
+    // Use the smallest entry in widths.
     const resourceWidth = useMemo(() => {
-        if (propResourceWidth) {
-            return propResourceWidth;
+        if (!widths) {
+            return undefined;
```

Comment:
Can this not be any default value returned if widths is not provided?

Reply:
It is odd that this would be `undefined`. I would assume that the components that care about this require it, right? We definitely don't want `resourceImage` throwing when it attempts to get `widths[index]` [here](https://github.com/magento/pwa-studio/blob/44965d25147980b97308a267b4539b0c7ba42015/packages/peregrine/lib/talons/Image/useResourceImage.js#L44).

Maybe instead of having the `useImage` talon do this work you could just pass the `widths` array to the components that use it and have them do this work. We're already talking about potentially splitting them anyways so that components can directly import resource or simple image rather than `Image`.

Reply:
This is a little shortcut if the caller doesn't pass a `width` but does pass `widths`.

If they don't pass either one, we literally don't have any information to go off of - `undefined` seemed the best choice here. This will result in the `img` element rendered to the DOM to not have a `width` attribute and the network request to fetch this image won't include resize parameters.

Reply:
I realized it wasn't preferring the passed-in `width` so I updated that in https://github.com/magento/pwa-studio/pull/1956/commits/4e798f71e9c0329caae579550104cb0ae400ea88.